src/TinyIoC/TinyMessenger.cs

```diff
@@ -781,12 +784,23 @@ where sub.Subscription.ShouldAttemptDelivery(message)
             });
         }
 
+        #if !NETSTANDARD
         private void PublishAsyncInternal<TMessage>(TMessage message, AsyncCallback callback) where TMessage : class, ITinyMessage
         {
             Action publishAction = () => { PublishInternal<TMessage>(message); };
 
             publishAction.BeginInvoke(callback, null);
         }
+        #else
+        private async void PublishAsyncInternal<TMessage>(TMessage message, AsyncCallback callback) where TMessage : class, ITinyMessage

```

Comment:
Is there any reason for adding this?

Reply:
Yes. I noted it in the message for the commit where I added it, but I can elaborate. Here's the commit message:

> Fix a .NET Core incompatibility with Delegate.BeginInoke in TinyMessenger (see https://devblogs.microsoft.com/dotnet/migrating-delegate-begininvoke-calls-for-net-core/).  Now all tests pass.

This is actually the change that made me want to get started on this PR in the first place. .NET Core doesn't work with BeginInvoke (the IAP model is supplanted by the TAP model) and this method wasn't working.  

So the else case works for .NET Core (or other runtimes that support .NET standard).  The link to the Microsoft devblog has all of the nitty gritty details.